packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -163,24 +209,30 @@ export class ChainService implements ChainServiceInterface {
 
     const captureExpectedErrors = async (reason: any) => {
       if (reason.error?.message.includes('Channel finalized')) {
+        this.logger.warn(
+          {channelId, determinedBy: 'Revert reason'},
+          'Transaction to conclude channel failed: channel is already finalized'
+        );
         return;
       }
-      const firstState = finalizationProof[0];
-      const [, finalizesAt] = await this.nitroAdjudicator.getChannelStorage(
-        getChannelId({
-          ...firstState,
-          participants: firstState.participants.map(p => p.signingAddress),
-        })
+
+      const [, finalizesAt] = await this.nitroAdjudicator.getChannelStorage(channelId);
+
+      const {timestamp: latestBlockTimestamp} = await this.provider.getBlock(
+        await this.provider.getBlockNumber()
       );
 
-      const latestBlockTimestamp = (
-        await this.provider.getBlock(await this.provider.getBlockNumber())
-      ).timestamp;
       // Check if the channel has been finalized in the past
-      if (latestBlockTimestamp >= Number(finalizesAt)) return;
+      if (latestBlockTimestamp >= Number(finalizesAt)) {
+        this.logger.warn(
+          {channelId, determinedBy: 'Javascript check'},
+          'Transaction to conclude channel failed: channel is already finalized'
+        );
```

Comment:
A return is missing. I am surprised unit tests didn't fail...

Reply:
I think it is always covered by the revert reason. TBH I don't know why do this check twice

Reply:
Now that I think about it, the revert reason is not present on Rinkeby: https://github.com/ethers-io/ethers.js/issues/446. So I think we need the return.

Unit tests are not failing because we run the tests against Ganache.